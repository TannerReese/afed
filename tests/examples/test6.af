# Test matrix arithmetc

M "str" = ``,
M[6, -1] = ``,
M[[0, 1], [2]] = ``,

trsp m1 = ``,
m3 * m2 = ``,
m2 * m1 = ``,
4 * m1 + trsp m2 = ``,
m1 * m2 - m3 / 7 = ``,

vc: V[1, 2, -3],
vc1: m1 * vc = ``,
vc2: vc * m2 = ``,
m3 * (vc1 + vc2) = ``,
m2 1 = ``,

m1: M[[-4, 0.1, 6/7], [-2/13, 0, 8]],
m2: mat.M[[1, 2], [-4, 3/5], [2/7, -0.8]],
m3: M[[4, 5], [-3, 13]],

