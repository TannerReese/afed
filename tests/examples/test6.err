line 3, column 11 Eval Error: Expected array, but found string
line 4, column 12 Eval Error: Expected array, but found number
line 5, column 18 Eval Error: Matrix cannot have jagged rows
line 8, column 11 Eval Error: For matrix multiplication, 2 and 3 do not match
4 Eval Errors encountered
