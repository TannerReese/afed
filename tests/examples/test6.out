# Test matrix arithmetc

M "str" = `Eval Error: Expected array, but found string`,
M[6, -1] = `Eval Error: Expected array, but found number`,
M[[0, 1], [2]] = `Eval Error: Matrix cannot have jagged rows`,

trsp m1 = `M[[-4, -2 / 13], [0.1, 0], [6 / 7, 8]]`,
m3 * m2 = `Eval Error: For matrix multiplication, 2 and 3 do not match`,
m2 * m1 = `M[[-56 / 13, 0.1, 118 / 7], [1034 / 65, -0.4, 48 / 35], [-1.0197802197802197, 0.02857142857142857, -6.155102040816327]]`,
4 * m1 + trsp m2 = `M[[-15, -3.6, 26 / 7], [18 / 13, 3 / 5, 31.2]]`,
m1 * m2 - m3 / 7 = `M[[-4.726530612244898, -9.34], [233 / 91, -8.564835164835165]]`,

vc: V[1, 2, -3],
vc1: m1 * vc = `V[-6.371428571428572, -314 / 13]`,
vc2: vc * m2 = `V[-55 / 7, 5.6000000000000005]`,
m3 * (vc1 + vc2) = `V[-149.68351648351648, -198.5142857142857]`,
m2 1 = `V[-4, 3 / 5]`,

m1: M[[-4, 0.1, 6/7], [-2/13, 0, 8]],
m2: mat.M[[1, 2], [-4, 3/5], [2/7, -0.8]],
m3: M[[4, 5], [-3, 13]],

